Adapter/PhpcrOdmAdapter.php

```diff
@@ -137,12 +138,24 @@ public function createAutoRoute(UriContext $uriContext, $contentDocument, $autoR
             $parentDocument = $document;
         }
 
+        $finalAutoRoutePath = $path . '/' . $headName;
+        $node = $this->dm->find(null, $finalAutoRoutePath);
+        if ($node) {
+            if ($node instanceof Generic) {
+                return $this->convertGenericNodeInAutoRouteNode($node, $contentDocument, $autoRouteTag, $routeType);
+            }
+            $nodeClass = get_class($node);
+            $genericFqcn = 'Doctrine\ODM\PHPCR\Document\Generic';
+            throw new \RuntimeException(
+                "Unexpected node class '$nodeClass' at path '$finalAutoRoutePath'. Only '$genericFqcn' expected."
```

Comment:
use `sprintf`, and perhaps reword as follows:

````
sprintf(
    'Encountered existing PHPCR-ODM document at path "%s" of class "%s", the route tree should contain only instances of AutoRouteInterface.'
    $path,
    $documentClass
`)
````

Note that documents that are returned as `Generic` actually are not managed by PHPCR-ODM - the nodes exist in the tree without any ODM mapping.